nightguard WatchKit Extension/InterfaceController.swift

```diff
@@ -437,5 +454,30 @@ class InterfaceController: WKInterfaceController, WKCrownDelegate {
         
         return "Snoozed " + String(AlarmRule.getRemainingSnoozeMinutes()) + "min"
     }
+    
+    func sendNightSafeRequest() {
+        
+        // request phone settings for determining to show the night safe indicator
+        RequestNightSafeMessage().send { [weak self] (response: ResponseNightSafeMessage) in
+            dispatchOnMain {
+                guard let self = self else { return }
+                guard self.isActive else { return }
+                
+                // show the night safe indicator if phone is active & lock screen is ON
+                self.nightSafeIndicator.setHidden(
+                    !(response.value.isPhoneActive && response.value.isScreenLockActive)
+                )
+                self.nightSafeIndicator.setAlpha(CGFloat(response.value.volumeLevel))
+                
+                delay(2) { [weak self] in
+                    guard let self = self else { return }
```

Comment:
This is the feature I called "night safe", it is simply an indicator on the watch (a green dot on the bottom), visible for two seconds (the idea was to be as non-intrusive as possible) if the phone is active and has the screen locked, and the sound volume is audible. The dot alpha value is the same as the volume level (for e.g. 50% alpha is volume is 50%), was a quick implementation (but probably not the best) and fine for a first version & concept check.
Now because the "no data" alarm can be deactivated, I should check it also for deciding on the "night safe" value.

Reply:
I meant the 

`guard let self = self`

tweak. I haven't thought about weak references too much but did some reading:
https://benscheirman.com/2018/09/capturing-self-with-swift-4-2/

That is something I don't like about swift. The second thing is the main/background thread stuff  that adds a lot of complexity. But that has nothing to do with your code ;-).

Reply:
Yep. That's one difference between Swift with ARC and Java with Garbage Collection; swift cannot solve common circular reference problems like Java does, but it is still very ok with ARC. I worked more than 15 years in C++ and I had to do all the memory management by hand (or using some language idioms). 
I confess that I am obsessed with capturing of self in closures, our team had so many memory leaks & bugs that costed us over $100K because of some missing [weak self]; nightguard is a small app, but leaks can be tricky also here. So, I always check the context of self: if is in a closure, I'll avoid capturing it with weak/unowned.

Reply:
Yes - that seems to be very expensive and error prone. It's although crazy how many app crashes we have for such a simple app. I think that increased after introducing background updates. But that's just a guess. From java we are using exception barriers to stop apps from crashing. But I can't see something similar on the swift side here.